content/spin/distributing-apps.md

```diff
@@ -1,104 +1,123 @@
 ```
 
-Let's push the application from the [quickstart](./quickstart.md) to the registry:
+> Remember that if the artifact is private you will need to be logged in, with permission to access it.
 
-<!-- @selectiveCpy -->
+### Pulling a Published Application
+
+`spin up` automatically downloads the application from the registry. If you want to manually download the application, without running it, use the `spin registry pull` command:
+
+<!-- @nocpy -->
 
 ```bash
-$ export BINDLE_URL=http://localhost:8000/v1
-$ spin bindle push --file spin.toml
-pushed: spin-hello-world/1.0.0
+$ spin registry pull ghcr.io/alyssa-p-hacker/hello-world:v1
 ```
 
-Now we can run the application using `spin up` directly from the registry:
+> Downloaded applications are cached. When run, or pulled again, Spin checks to see if they have changed from the cached copy, and downloads only the changes if any.
 
-<!-- @selectiveCpy -->
+## Signing Spin Applications and Verifying Signatures
+
+Because Spin uses the container registry standards to distribute applications, it can also take advantage of tooling built around those standards.  Here is an example of using [Cosign and Sigstore](https://docs.sigstore.dev/cosign/overview/) to sign and verify a Spin application:
+
+<!-- @nocpy -->
 
 ```bash
-$ spin up --bindle spin-hello-world/1.0.0
-```
+# Push your Spin application to any registry that supports the OCI registry artifacts,
+# such as the GitHub Container Registry, Docker Hub, Azure ACR, or AWS ECR.
+$ spin registry push ghcr.io/alyssa-p-hacker/hello-world:v1
+
+# You can now sign your Spin app using Cosign (or any other tool that can sign
+# OCI registry objects).
+$ COSIGN_EXPERIMENTAL=1 cosign sign ghcr.io/alyssa-p-hacker/hello-world@sha256:06b19
+Generating ephemeral keys...
+Retrieving signed certificate...
+tlog entry created with index: 12519542
+Pushing signature to: ghcr.io/alyssa-p-hacker/hello-world
 
-> A known error when running an application (with a high number of static assets) from Bindle on macOS is around [too many open files](https://github.com/fermyon/spin/issues/180). This issue can be worked around by setting a higher `ulimit -n` for the current shell
-> session.
+# Someone interested in your application can now use Cosign to verify the signature
+# before running the application.
+$ COSIGN_EXPERIMENTAL=1 cosign verify ghcr.io/alyssa-p-hacker/hello-world@sha256:06b19
+Verification for ghcr.io/alyssa-p-hacker/hello-world@sha256:06b19 --
+The following checks were performed on each of these signatures:
+  - The cosign claims were validated
+  - Existence of the claims in the transparency log was verified offline
+  - Any certificates were verified against the Fulcio roots.
 
-The application can also be prepared in a local directory before pushing to the registry by running `spin bindle prepare`.
\ No newline at end of file
+# The consumer of your app can now run it from the registry.
+$ spin up -f ghcr.io/alyssa-p-hacker/hello-world:v1
```

Comment:
@radu-matei are you happy for this to go out - I have reservations about implying the mutable tag is now verified but maybe `cosign` is doing something cunning to make it safe?

Reply:
Had a word with Radu and he wants this to go out as is, until we have pull by digest at which point we can update it.